### DISCUSSION TITLE
Sending commands not working

### DISCUSSION THREAD

**Discussion by hanabal2** [llamafilm/tesla-http-proxy-addon#58] [Q&A]
Hi, 

I finally got the http proxy working and home assistant can get all the data from the car (I think). but when I try to send any command, it doesnt work. Here is a copy of the log with the relevant error. please let me know if any other lines could be helpful. 

2024-03-24T10:16:21+13:00 [warn ] [708757aac0c3a5a5cefbdc07bd4de5bd] Discarding unauthenticated session info
2024-03-24T10:16:21+13:00 [warn ] [708757aac0c3a5a5cefbdc07bd4de5bd] Session info error: BadParameter: invalid public key
2024-03-24T10:16:21+13:00 [error] Returning error Internal Server Error

The car does wake up if it was sleeping. but doesnt execute the command. If the car was already awake, it also doesnt execute it.

Is there something ive done wrong that would cause the invalid public key?

Version: 1.3.7

**Comment by llamafilm** [comment]
I have not seen this error before but the issue seems clear from the error message: `invalid public key`.  What do you get when trying to hit your public key endpoint externally?  Part of the code in this add-on validates your public key, so you should have something similar to this in the log if it's working correctly:
```
HTTP/2 200 
server: nginx
date: Sat, 23 Mar 2024 21:36:19 GMT
content-type: text/plain
content-length: 178
last-modified: Sat, 23 Mar 2024 20:02:04 GMT
etag: "65ff353c-b2"
strict-transport-security: max-age=31536000; includeSubDomains
accept-ranges: bytes

-----BEGIN PUBLIC KEY-----
MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAE8zKbXEFKzp4YN/jZGUNeWbvpW2bB
skp4IfGFjmJPDXtRwiY2mXv5DKckiXC0H4rGlu6smtq7OT1HRsULU9RTMA==
-----END PUBLIC KEY-----
```

**Comment by hanabal2** [comment]
thanks for replying. Yeah when i hit the public key site as per the instructions, it loads just fine. I initially had a cert error but fixed that. 

Here is my log when restarting the proxy addon:
s6-rc: info: service s6rc-oneshot-runner: starting
s6-rc: info: service s6rc-oneshot-runner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service legacy-cont-init: starting
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service legacy-services: starting
s6-rc: info: service legacy-services successfully started
[11:13:52] INFO: Found existing keypair
[11:13:52] INFO: Testing public key...
HTTP/2 200 
server: nginx
date: Sat, 23 Mar 2024 22:13:52 GMT
content-type: text/plain
content-length: 178
last-modified: Sat, 23 Mar 2024 07:18:24 GMT
etag: "65fe8240-b2"
strict-transport-security: max-age=31536000; includeSubDomains
accept-ranges: bytes

-----BEGIN PUBLIC KEY-----
MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEFjpFmPq/Pzn7CvpXpTeQkVhbFg1z
dqukRdy4eEsULd7ViSEJDQNxcSHIndBNi4NmB39bJtZFZgs5a3P2m+gO8Q==
-----END PUBLIC KEY-----
[11:13:52] INFO: Starting Tesla HTTP Proxy

Do not listen on a network interface without adding client authentication. Unauthorized clients may
be used to create excessive traffic from your IP address to Tesla's servers, which Tesla may respond
to by rate limiting or blocking your connections.
2024-03-24T11:13:52+13:00 [debug] Creating proxy
2024-03-24T11:13:52+13:00 [info ] Listening on 0.0.0.0:443

I have also tried regenerating the tesla auth via the addon. while it seems to work fine and does as per the instructions, (even turning itself off) i keep hitting this error

**Comment by llamafilm** [comment]
Did you enroll public key in your vehicle?  When you click that button in the web ui, it should launch the Tesla iOS app and prompt you to enroll it.

**Comment by hanabal2** [comment]
Yes. I used the Android app but I clicked the enrol button and it loaded
the Tesla app and I saw my car. I assume that's all I needed to do.

**Comment by hanabal2** [comment]
Ahh I didn't get any notification to enrol. Was that what I am missing. I
wonder why I didn't get one. Does it only work on iOS?

**Comment by llamafilm** [comment]
After clicking that link on the Web UI page you should get a prompt like this:
![IMG_7915F37E8473-1](https://github.com/llamafilm/tesla-http-proxy-addon/assets/3991046/87b0bab1-fa90-4a3f-a6cf-2df499b2b43e)

I think you're the first person to try it on Android, but the [documentation](https://github.com/teslamotors/vehicle-command) says it should work the same.

**Comment by hanabal2** [comment]
Yeah I don't get that. It just loaded the app as if I pressed the icon on
my home screen.

I'll see if I can find an iOS device that I can use

**Comment by llamafilm** [comment]
You can also try typing that `_ak` URL into any note-taking app where it turns into a hyperlink and then click it.

**Comment by hanabal2** [comment]
Ok I got it working.

Turns out I needed my phone to be in range of the car when I authorised it.

**Reply by llamafilm** [reply]
Interesting!  That was not the case for me.
Are you able to send commands to the car from this phone when you're out of bluetooth range?

**Comment by hanabal2** [comment]
Yeah, I can do that.

I'll keep testing things and let you know if it doesn't work
